hammurabi/rules/text.py

```diff
@@ -310,9 +310,16 @@ def __get_target_match(self, lines: List[str]) -> List[str]:
         """
 
         target_match = list(filter(self.target.match, lines))
+        text = list(filter(lambda l: l.strip() == self.text, lines))
 
-        if not any(target_match):
-            raise LookupError(f'No matching line for "{self.target}"')
+        if target_match and text:
+            raise LookupError(f'Both "{str(self.target)}" and "{self.text}" exists')
+
+        if text:
```

Comment:
is it possible that `not target_match` is True here? According to the doc above, there should be an exception.
It is also hard to follow what should happen with the 3 ifs after each other (like why exactly is this the order, what if we need to add any code in this function) - is it possible to refactor them into a more self explanatory version?

Reply:
> is it possible that not target_match is True here? According to the doc above, there should be an exception.

Yes it is, in the case if the given `text` (the line which will replace the original line) is not in the file yet and the `target_match` (the line which should be replaced) is not in the file either. In that case we can not replace the expected line and we have no line already replaced. This scenario should result in an exception since if some rule is depending on the replacement, they would fail.